src/components/questions/addAnswer.js

```diff
@@ -0,0 +1,147 @@
+import React from "react";
+import axios from "axios";
+
+class AddAnswer extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      qID : this.props.qID,
+      answer:"",
+      nickname:"",
+      email:"",
+      requiredAnswer: false,
+      requiredNickname: false,
+      requiredEmail: false,
+      validEmail: false
+    }
+    this.onChange = this.onChange.bind(this);
+    this.submitAnswer = this.submitAnswer.bind(this);
+    this.checkEmailValidity = this.checkEmailValidity.bind(this);
+  }
+
+  onChange(event){
+    console.log(event.target.value);
+    let feild = event.target.name;
+    let value = event.target.value;
+    this.setState({[feild] : value})
+  }
+
+  checkEmailValidity(email){
+    let regex = RegExp(/^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/);
```

Comment:
You will have to explain this to me one day.

Reply:
I am also unclear on how the test method works.

Reply:
mayanne is the Regex expert! 😂